### PR TITLE
Removed using namespace std from all files

### LINKED ISSUE #35713: [Build] Unnecessary `using namespace std` creates ambiguity error to C++23 standard compiler.
### Build issue(s)

When compiling both sources `connectedhomeip\src\app\clusters\refrigerator-alarm-server\refrigerator-alarm-server.cpp` and `connectedhomeip\src\app\clusters\mode-select-server\mode-select-server.cpp`, an error of ambiguity pops out by the compiler regarding the use of `to_underlying()`.

Solution: Remove the line `using namespace std;` [here](https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp#L35) and [here](https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/mode-select-server/mode-select-server.cpp#L40).

### Platform

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jadhavrohit924**
### Problem
- Fixes #35713.

### Changes
- Removed using namespace std from all sources.

### Test
- Build all-cluster-app and tv-app for Linux.

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by andy31415 — CHANGES_REQUESTED** [review]
The description in https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/TypeTraits.h#L32 seems to say we intentionally overloaded this to have C++23 shims.

Don't we have other alternatives? i.e. for C++23 just use the existingimplementation from the standard instead of implementing our own.

This PR seems to propose to have the workaround forever instead of treating it as a temporary shim.

File: src/lib/support/SafeInt.h [hunk 1]
```diff
@@ -37,11 +37,10 @@ namespace chip {
 template <typename T, typename U, std::enable_if_t<std::is_integral<T>::value, int> = 0>
 bool CanCastTo(U arg)
 {
-    using namespace std;
```
Comment (bzbarsky-apple) [inline src/lib/support/SafeInt.h:40]: Probably OK, but this use was safe: it's scoped to this function, and this function does not use `to_underlying`.

File: src/lib/support/SafeInt.h [hunk 2]
```diff
@@ -127,10 +126,9 @@ bool CanCastTo(U arg)
 template <typename T>
 typename std::enable_if<std::is_unsigned<T>::value, typename std::make_signed<T>::type>::type CastToSigned(T arg)
 {
-    using namespace std;
```
Comment (bzbarsky-apple) [inline src/lib/support/SafeInt.h:130]: Again, this use was fine.

File: src/lib/support/TypeTraits.h [hunk 3]
```diff
@@ -26,8 +26,21 @@
...
+#if __has_include(<utility>) // For C++23 and later, include <utility> if available
+#include <utility>           // Contains std::to_underlying
+#endif
+
 namespace chip {
 
+#if __cplusplus >= 202300L
+
+template <class T>
+constexpr auto to_underlying(T e)
```
Comment (bzbarsky-apple) [inline src/lib/support/TypeTraits.h:38]: Does `using to_underlying = std::to_underlying;` or similar not work?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/lib/support/SafeInt.h (hunk 1 above)
Comment (jadhavrohit924) [reply src/lib/support/SafeInt.h]: Restored it.

File: src/lib/support/SafeInt.h (hunk 2 above)
Comment (jadhavrohit924) [reply src/lib/support/SafeInt.h]: Restored

File: src/lib/support/TypeTraits.h (hunk 3 above)
Comment (jadhavrohit924) [reply src/lib/support/TypeTraits.h]: Right. Done

**Comment by jadhavrohit924** [comment]
@andy31415 I have addressed the changes please take a look.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**